lib/Status.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Nominatim;
+
+use Exception;
+use PEAR;
+
+class Status
+{
+    protected $oDB;
+
+
+    public function __construct(&$oDB)
+    {
+        $this->oDB =& $oDB;
+    }
+
+    // return null on success
+    public function status()
+    {
+        if (!$this->oDB || PEAR::isError($this->oDB)) {
+            return 'No database';
+        }
+
+        $sStandardWord = $this->oDB->getOne("SELECT make_standard_name('a')");
+        if (PEAR::isError($sStandardWord)) {
+            return 'Module failed';
+        }
+
+        if ($sStandardWord != 'a') {
+            return 'Module call failed';
+        }
+
+        $sSQL = 'SELECT word_id, word_token, word, class, type, country_code, ';
+        $sSQL .= "operator, search_name_count FROM word WHERE word_token IN (' a')";
+        $iWordID = $this->oDB->getOne($sSQL);
+        if (PEAR::isError($iWordID)) {
+            return 'Query failed';
+        }
+        if (!$iWordID) {
+            return 'No value';
+        }
+
+        return;
+    }
+
+    public function dataDate()
+    {
+        $sSQL = "SELECT EXTRACT(EPOCH FROM lastimportdate - '2 minutes'::interval) FROM import_status LIMIT 1";
```

Comment:
We probably should get rid of the 2 min hack as well at this point. The factor is there because the data from the last update step might not have been indexed yet and therefore cannot be found. However, indexing can take arbitrarily long so there is no telling how far back the last update step was completed.